DatadogRUM/Sources/Integrations/TelemetryReceiver.swift

```diff
@@ -203,7 +204,8 @@ internal final class TelemetryReceiver: FeatureMessageReceiver {
 
         rum.eventWriteContext { context, writer in
             // reset recorded events on session renewal
-            let attributes: [String: String]? = context.featuresAttributes["rum"]?.ids
+            let rumAttributes: [String: String?]? = context.featuresAttributes[RUMFeature.name]?.ids
+            let attributes = rumAttributes?.compactMapValues { $0 }
             let sessionId = attributes?[RUMContextAttributes.IDs.sessionID]
 
             if sessionId != self.currentSessionID {
```

Comment:
I'm not fully sure on what was the impact here. If no `action.id` was present in RUM context, then `sessionId` was always `nil`. Looking at further logic, I think we might have been losing significant volume of telemetries because `false` was dominant in `sessionId != self.currentSessionID` comparison (`nil != nil`). That led to never reseting the `eventIDs` buffer, hence always being capped by `TelemetryReceiver.maxEventsPerSessions`.